### PR TITLE
Requiring valid title for menu items

### PR DESCRIPTION
**Pull request by thebeuving**
Requiring more than a space to create a valid title

### CODE REVIEW

**Comment by brianteeman** [comment]
This is a valid issue BUT as we are in language string freeze I fear it may have to wait until after 3.4

@wilsonge is that true
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/5974">issues.joomla.org/joomla-cms/5974</a>.</sub>

File: administrator/language/en-GB/en-GB.com_menus.ini [hunk 1]
```diff
@@ -18,6 +18,7 @@ COM_MENUS_ERROR_ALL_LANGUAGE_ASSOCIATED="A menu item set to All languages can't
 COM_MENUS_ERROR_ALREADY_HOME="Menu item already set to home."
 COM_MENUS_ERROR_MENUTYPE="Please change the Menu type. The terms 'menu' and 'main' are reserved for Backend usage."
 COM_MENUS_ERROR_ONE_HOME="Only one menu item can be a home link for each language."
+COM_MENUS_WARNING_PROVIDE_VALID_TITLE="Please provide a valid, non-blank title."
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.com_menus.ini]: Can you move this line please so that it listed in the file in alphabetical ordering

**Comment by brianteeman** [comment]
@test issue observed and problem resolved by PR<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/5974">issues.joomla.org/joomla-cms/5974</a>.</sub>

File: administrator/language/en-GB/en-GB.com_menus.ini (hunk 1 above)
Comment (thebeuving) [inline administrator/language/en-GB/en-GB.com_menus.ini]: Fixed

**Comment by wilsonge** [comment]
I'm afraid so. No new strings from now unless absolutely essential. As this is an old bug it can wait for after 3.4

**Comment by thebeuving** [comment]
Alright that makes sense, does that mean I just need to resubmit the pull request to staging branch after Joomla 3.4 is released, or will it automatically get re-evaluated at that time?

**Comment by wilsonge** [comment]
We'll take care of it when 3.4 is released. But please give us a prod by mailing on here if it looks like we've forgotten :)

**Comment by brianteeman** [comment]
Set the Milestone to 3.4.1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/5974">issues.joomla.org/joomla-cms/5974</a>.</sub>

File: libraries/legacy/table/menu.php [hunk 2]
```diff
@@ -90,6 +90,13 @@ public function bind($array, $ignore = '')
 	 */
 	public function check()
 	{
+		//Make sure there is a valid title, a space doesn't count
```
Comment (zero-24) [inline libraries/legacy/table/menu.php]: @thebeuving can you add one space after the `//`? see: https://travis-ci.org/joomla/joomla-cms/jobs/49486536

Comment (thebeuving) [inline libraries/legacy/table/menu.php]: Fixed.

**Comment by infograf768** [comment]
As this PR modifies the library, please change the new string to a JLIB one and add it in both en-GB.lib_joomla.ini (front and back)
Example of existing string in core:
JLIB_DATABASE_ERROR_MENU_UNIQUE_ALIAS

**Comment by nonumber** [comment]
Yes, language string should be named:
```JLIB_DATABASE_ERROR_MUSTCONTAIN_A_TITLE_MENU```

See the category.php and module.php as example:
```
		// Check for a title.
		if (trim($this->title) == '')
		{
			$this->setError(JText::_('JLIB_DATABASE_ERROR_MUSTCONTAIN_A_TITLE_CATEGORY'));

			return false;
		}
```
You could also keep the comments the same.

**Comment by thebeuving** [comment]
Ok I went ahead and made those adjustments.

**Comment by nonumber** [comment]
@test all good

**Comment by zero-24** [comment]
RTC as we have two successful tests. Thanks!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/5974">issues.joomla.org/joomla-cms/5974</a>.</sub>

**Comment by wilsonge** [comment]
Merged! Thanks!

**Pull request merged**